### ISSUE TITLE
Scroll problem if more than 199 elements

### ISSUE THREAD

**Issue by tcivera** [wenzhixin/multiple-select#552]
When the select has more than 199 elements/options, the vertical scroll stops to work.

**Comment by tcivera** [comment]
The problem seems to be in the VirtualScroll class. Commenting the line _this.initDOM(_this.rows) inside the onScroll function, seems to work correctly:

var onScroll = function onScroll() {
        if (_this.lastCluster !== (_this.lastCluster = _this.getNum())) {
          //_this.initDOM(_this.rows);

          _this.callback();
        }
      };

**Comment by the-jiveman** [comment]
I can confirm the same problem on my end. Populating the array with 198 items or less, there is no scroll issue.

**Comment by the-jiveman** [comment]
@tcivera , If I comment out the `_this.initDOM(_this.rows);` line, it does get rid of the scroll/flicker problem, but only 199 items are shown in the menu now.

**Comment by tcivera** [comment]
Thanks 'the-jiveman' for the solution. Sorry, I didn't realize that with commenting the line only the first 199 items were displayed.
I have the select in a hidden tab, so, it works adding the refresh when the first time the tab is shown.

It seems that problem is that the height is not calculated correctly in the getRowsHeight() function  in the VirtualScroll class:

```
          var nodes = this.contentEl.children;
          var node = nodes[Math.floor(nodes.length / 2)];          
          this.itemHeight = node.offsetHeight;  //The offsetHeight returns 0 because the item (li) is not visible.
```

**Comment by the-jiveman** [comment]
Yes @tcivera, I found the same lines of code to be the culprit, but didn't have the time to try and fix it at the source, as I needed to move ahead with my project. I really want to use Multiple Select in my project, as it's the cleanest looking and well documented multi-select control I found, so I hope the real fix comes coon. In the meantime, I may have to deal with it using the `refresh` workaround.

**Comment by cantak** [comment]
Commenting out "_this.initDOM(_this.rows);" works best for me. Many of these components are sitting on a popup modal so they are never showing when the page loads.

**Comment by ghiscoding** [comment]
FYI, this is probably caused by the Virtual Scroll which used to be 200 but was later increased to 2000 in this commit a3d87566 , also worth to know that the v-scroll is calculated from `CLUSTER_BLOCKS * BLOCK_ROWS` which is now equal to 2000. So I'm assuming that you no longer have the problem since the increase

https://github.com/wenzhixin/multiple-select/blob/cd6c057769970efccda2c4ba4f46d19bdc83dfe9/src/constants/index.js#L2-L3

**Comment by yoreek** [comment]
I found a problem in VirtualScroll when dividing by zero. After fix https://github.com/wenzhixin/multiple-select/pull/627 it works correctly.

**Comment by ashishchawda17** [comment]
@ghiscoding I'm having as use case where I've more than 2000 items(take 2700 for example) in the dropdown, it does not show items after 1999th index
I'm getting data from an API call, adding `option` elements dynamically to `select` element and then initializing the multipleSelect instance
This issue was noticed between 1.5.2 and 1.7.0.
Does @the-jiveman's workaround suggestion fix it too?

**Comment by wenzhixin** [comment]
Fixed in the latest version.
